lib/addons/adot/index.ts

```diff
@@ -47,18 +47,27 @@ export class AdotCollectorAddOn extends CoreAddOn {
     }
     @dependable(CertManagerAddOn.name)
     deploy(clusterInfo: ClusterInfo): Promise<Construct>  {
-
-        const cluster = clusterInfo.cluster;
-
         if (semverComparator("0.88",this.coreAddOnProps.version)) {
             console.log("Used Adot Addon Version is Valid");
         } 
         else {
             throw new Error(`Adot Addon Version is not Valid and greater than 0.88.0`);
         }
+        
+        const addOnPromise = super.deploy(clusterInfo);
+        return addOnPromise;
+    }
 
+    /**
+     * Overriding base class method to create namespace and register permissions.
+     * @param clusterInfo 
+     * @param name 
+     * @returns 
+     */
+    createNamespace(clusterInfo: ClusterInfo, namespaceName: string): IConstruct | undefined {
```

Comment:
The name `createNamespace` is misleading because we also deploying OTEL permission Manfest here. Can we move that out?

Reply:
Not for ADOT. I am using a pattern called template method. A noop (abstract) method named `createNamespace` is defined in the core addon. Core addon uses this method to create namespace. ADOT overrides this method to create a namespace and handle other manifest that must be created with the namespace. Since the base template method is defined in the core addon, it has to use a generic name.

Reply:
Makes total sense i saw that, I would rather rename the `createNamespace` to something meaningful like `createAddonResources`. But approach is good!

Reply:
I see your logic, but i would rather keep the name of the method as `createNamespace` to address 99% of cases and override it for adot which has a design peculiarity that adds extra compexity, than make the name like createAddonResources which should probably return an array of constructs.